src/svm/interface/svm.c

```diff
@@ -2257,6 +2335,78 @@ PetscErrorCode SVMComputeModelParams(SVM svm)
   PetscFunctionReturn(0);
 }
 
+#undef __FUNCT__
+#define __FUNCT__ "SVMLoadMatGramian"
+/*@
+  SVMLoadMatGramian - Loads precomputed Gramian (kernel) matrix.
+
+  Collective on PetscViewer
+
+  Input Parameters:
++ svm - SVM context
+- v - viewer
+
+ Level: intermediate
+
+.seealso SVM, SVMViewMatGramian(), SVMLoadTrainingDataset()
+@*/
+PetscErrorCode SVMLoadMatGramian(SVM svm,PetscViewer v)
+{
+  MPI_Comm   comm;
+  PetscBool  view_io,view_dataset;
+
+  const char *dataset_file;
+
+  PetscFunctionBeginI;
+  PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
+  PetscValidHeaderSpecific(v,PETSC_VIEWER_CLASSID,2);
+
+  PetscLogEventBegin(SVM_LoadGramian,svm,0,0,0);
+  if (svm->ops->loadgramian) {
+    TRY( svm->ops->loadgramian(svm,v) );
+  }
```

Comment:
error out when implementation is not available?

Reply:
Great! I think it should be useful to implement an error message in this case. However, I followed the PETSc and PERMON(QP) implementations of similar functions, and I didn't see it. I support this idea, however, not in this PR. I'll create an issue associated with your idea.